manager/notifier.go

```diff
@@ -231,6 +231,70 @@ func (n *notifier) sendHipChatNotification(op notificationOp, config *pb.HipChat
 	return nil
 }
 
+func (n *notifier) sendSlackNotification(op notificationOp, config *pb.SlackConfig, a *Alert) error {
+	// https://api.slack.com/incoming-webhooks
+	incidentKey := a.Fingerprint()
+	color := ""
+	status := ""
+	switch op {
+	case notificationOpTrigger:
+		color = config.GetColor()
+		status = "firing"
+	case notificationOpResolve:
+		color = config.GetColorResolved()
+		status = "resolved"
+	}
+
+	buf, err := json.Marshal(map[string]interface{}{
+		"channel": config.GetChannel(),
+		"attachments": []interface{}{
+			map[string]interface{}{
```

Comment:
How about a properly typed struct here? Even though more verbose, I always prefer those over maps and empty interfaces. Especially in bad cases of nesting like this one.

I know the other implementations are not leading by best example - but we can start improving here.